File: Modix/ClientApp/src/models/GuildStatApiData.ts
import { PieChartItem } from '@/models/PieChart'

export default interface GuildStatApiData
{
    guildName: string;
    guildRoleCounts: GuildRoleCount[];
    topUserMessageCounts: PerUserMessageCount[];
}

export interface GuildRoleCount extends PieChartItem
{
}

export interface PerUserMessageCount
{
    username: string;
    discriminator: string;
    rank: number;
    messageCount: number;
    isCurrentUser: boolean;
}

File: Modix/Models/GuildStatApiData.cs
using System.Collections.Generic;
using Modix.Data.Models.Core;
using Modix.Services.GuildStats;

namespace Modix.Models
{
    public class GuildStatApiData
    {
        public string GuildName { get; set; }

        public List<GuildRoleCount> GuildRoleCounts { get; set; }

        public IReadOnlyCollection<PerUserMessageCount> TopUserMessageCounts { get; set; }
    }
}
